website/static/js/apiApplication.js

```diff
@@ -0,0 +1,271 @@
+ * like an update page.
+ */
+var ApplicationViewModel = function (urls) {
+    // Read and update operations
+
+    var self = this;
+    self.content = ko.observable(new ApplicationData);
+    self.message = ko.observable();
+    self.messageClass = ko.observable();
+
+    self.dataUrl = urls.dataUrl; // Use for editing existing instances
+    self.submitUrl = urls.submitUrl; // Use for creating new instances
+
+    if (self.dataUrl){ // Support detail views
+        self.fetch(self.dataUrl);
+    }
+};
+
+ApplicationViewModel.prototype.fetch = applicationFetch;
+
+ApplicationViewModel.prototype.updateApplication = function () {
+    // Update an existing application (has a known dataUrl) via PATCH request
+    var self = this;
+
+    var url = self.dataUrl;
+
+    var payload = self.content().toJSON();
+
+    var request = $osf.ajaxJSON("PATCH", url, {isCors: true, data: payload});
+
+    request.done(function (data) {
+        self.content().fromJSON(data.data);  // Update the data with what request returns- reflect server side cleaning
+        self.changeMessage(
+            "Application data submitted",  // TODO: Some pages (eg profile) show a one-line message for updates; others use a growl box. Current best practices?
+            "text-success",
+            5000);
+    });
+
+    request.fail(function (xhr, status, err) {
+            $osf.growl('Error', 'Could not send request. Please refresh the page and try ' +
+            'again or contact <a href="mailto: support@cos.io">support@cos.io</a> ' +
+            'if the problem persists.', 'danger');
+
+            Raven.captureMessage('Error updating instance', {
+                url: url,
+                status: status,
+                error: err
+            });
+    });
+};
+
+ApplicationViewModel.prototype.createApplication = function () {
+    // Create a new application instance via POST request (when model has submitUrl, but no dataUrl)
+    var self = this;
+    var payload = self.content().toJSON();
+
+    var url = self.submitUrl;
+
+    var request = $osf.ajaxJSON('POST', url, {isCors: true, data: payload});
+
+    request.done(function (data) {
+        self.content().fromJSON(data.data);  // Update the data with what request returns- reflect server side cleaning
+
+        // TODO: Window.location refreshes anyway, rendering some of this KO.js code possibly unnecessary. Either reload seamlessly in-page, or refresh; doing both is architecturally awkward.
+        // pushState is an alternative (and used in OSF search page), but not supported uniformly in older browsers
+        $osf.growl('Application created', 'Application created!', 'success');
+
+
+        // Update behaviors: after creation, this should look & act like a detail view for existing application
+        window.location = data.data.links.html; // Update address bar to show new detail page
+        self.dataUrl = data.data.links.self;
+    });
+
+    request.fail(function (xhr, status, err) {
```

Comment:
This looks fairly generic. Might want to go ahead and make it a function for osfHelpers. If you had a more specific growl error, then no, but as it stands, all that needs to happen is passing in a little context for Raven.